lib/devise/controllers/helpers.rb

```diff
@@ -9,6 +9,15 @@ module Helpers
       include Devise::Controllers::StoreLocation
 
       included do
+        # inside any controller the strategies that are consulted can be configured
+        # such that only whitelisted strategies are run. 
+        # Example:
+        #   self.strategies = :authenticatable
+        #   self.strategies = [:authenticatable, :database_authenticatable]
+        #   self.strategies = { user: :authenticatable, admin: :database_authenticatable }
+        #
+        class_attribute :strategies
```

Comment:
Thoughts on making this a controller instance method as opposed to a class attribute? Agreed that this would be useful -- I'm in the middle of implementing something similar for a multi-tenant app with a single scope but a few separate auth mechanisms -- but I think it could be helpful to have the request context available if possible.

I realize Warden encourages you to locate strategy eligibility in the strategy itself, but if we did want to pick explicitly in the controller based on the request or some runtime information, that would be possible with an instance method but not a class attribute.

Making this an instance method would also make this function a little closer to the way the `auth_options` work in `Devise::SessionsController#create.`: https://github.com/heartcombo/devise/blob/main/app/controllers/devise/sessions_controller.rb#L47

Reply:
That seems perfectly reasonable. I'll take a look.

Reply:
changed on your recommendation